libr/core/cbin.c

```diff
@@ -497,7 +501,10 @@ R_API void r_core_anal_type_init(RCore *core) {
 	const char *dir_prefix = r_config_get (core->config, "dir.prefix");
 	int bits = core->assembler->bits;
 	types = core->anal->sdb_types;
- 	// make sure they are empty this is initializing
+	// make sure they are empty and this is initializing
+	if  (sdb_count (types) != 0) {
```

Comment:
this !=0 is implicit you can remove that. `if (sdb_count (types) > 0) {` is better imho.. and i think in this case this is completely innefficient and we should add sdb_empty() is sdb_is_empty(). to just reuturn true in the first hit instead of having to interate over all the database